modules/javafx.controls/src/main/java/javafx/scene/control/Tooltip.java

```diff
@@ -170,6 +171,16 @@ public Tooltip(String text) {
         getStyleClass().setAll("tooltip");
     }
 
+    @Override
+    protected void show() {
+        // The very first show call is just for us to do the correct CSS processing, so we ignore the request here.
+        if (!cssForced) {
```

Comment:
I am *very* suspicious of this change.  Yes, it sort of works, but I wonder if it might backfire as it breaks the contract of show() and also depends on some assumptions ("The very first show call is just for us to do the correct CSS processing").

Would it be possible to try the applySceneStylesFromOwner() approach, maybe by moving the method to Utils to avoid changing the API?  This issue does not seem to be limited to Tooltip - there is a similar code in CustomColorDialog:88 for example.

What do you think?

Reply:
As written above, that would also be a possibility I explorer and that worked as well. Would be fine for me and I agree that this is rather unexpected. I will explore the other solution soon, when time permits.